woocommerce/woocommerce-gutenberg-products-block
# Interactivity API Collection Rating Filter

## What / Why

This includes:

1.  Implement MVP rating filter block as part of woocommerce/woocommerce#42229
2. New reusable interactivity checkbox list component
3. Fixes and improvements for reusable dropdown, including loading own styles and fixes to issues with context getting out of date.
4. Note that we don't support server rendered html in labels for dropdown component yet, will need a follow up refactor to support that and render actual stars HTML inside the dropdown.

## Testing Instructions

There are 2 modes for the rating, dropdown and checkbox list. You'll need to switch and test both types.

1. You'll need some products with different rating numbers, ideally a few of each rating type. Please note there is a bug in how we filter collection data right now that the counts of rated products may be incorrect so we're testing basically that the rating filter does some filtering when you choose different options and that the filtered products shown have the ratings you specified in the filter, but not that the correct counts for ratings are shown.
2. Insert the interactivity rating filter into collection filter block, try the dropdown and checkbox list mode and see that products are filtered when selecting options.


## Screenshots or screencast
<img width="658" alt="Screenshot 2023-12-04 at 5 01 12 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/1281828/fc032dd4-93b0-4739-b673-c82525abdee0">
<img width="649" alt="Screenshot 2023-12-04 at 5 00 57 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/1281828/d583dd99-b01d-460d-b8f8-60d221215459">

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Add suggested changelog entry here.
